src/wasi/Wasi.cpp

```diff
@@ -64,6 +64,19 @@ void WASI::proc_exit(ExecutionState& state, Value* argv, Value* result, Instance
     ASSERT_NOT_REACHED();
 }
 
+void WASI::random_get(ExecutionState& state, Value* argv, Value* result, Instance* instance)
+{
+    ASSERT(argv[0].type() == Value::I32);
+    ASSERT(argv[1].type() == Value::I32);
+    if (!WASI::checkMemOffset(instance->memory(0), argv[0].asI32(), argv[1].asI32())) {
+        result[0] = Value(WASI::wasi_errno::inval);
+    }
+
+    void* buf = (void*)(instance->memory(0)->buffer() + argv[0].asI32());
+    uvwasi_size_t length = argv[1].asI32();
+    result[0] = Value(uvwasi_random_get(WASI::m_uvwasi, buf, length));
```

Comment:
According to the [spec](https://github.com/WebAssembly/WASI/blob/main/legacy/preview0/docs.md#-random_getbuf-pointeru8-buf_len-size---result-errno),
`random_get` function writes a random data into a buffer input.
Does this buffer mean `memory`?

Reply:
I think yes, but I'll ask my co-worker about it to be sure.

Reply:
Yes, they do mean the memory.

https://github.com/WebAssembly/WASI/blob/main/legacy/application-abi.md
> Regardless of the kind, all modules accessing WASI APIs also export a linear memory with the name memory. Data pointers in WASI API calls are relative to this memory's index space.

Maybe in the future if multiple memory ever becomes a standard then these funcitons may also gain an index to which memory to use.